linera-chain/src/block_tracker.rs

```diff
@@ -149,14 +154,36 @@ impl<'resources> BlockExecutionTracker<'resources> {
             ))
             .with_execution_context(context)?;
 
+        // Account for blobs published by this transaction directly.
         for blob in &txn_outcome.blobs {
-            if blob.content().blob_type() == BlobType::Data {
+            resource_controller
+                .policy()
+                .check_blob_size(blob.content())
+                .with_execution_context(context)?;
+            if blob.content().blob_type().is_user_blob() {
                 resource_controller
                     .track_blob_published(blob.content())
```

Comment:
How about calling `track_blob_published(&blob)` and letting the resource controller decide the fee based on the blob type? Also it would probably make sense to charge most blobs in the same way very soon.

Reply:
I think we don't charge only for the `BlobType::Committee` which is a system-type blob required for correctness.